src/core/server/saved_objects/migrations/core/build_active_mappings.ts

```diff
@@ -142,6 +142,9 @@ function defaultMapping(): IndexMapping {
       namespace: {
         type: 'keyword',
       },
+      namespaces: {
+        type: 'keyword',
+      },
```

Comment:
I guess this is all but easy, but having these two fields sounds and the distinction between single and multi namespaces SOs sounds wrong to me. How hard and breaking would be to migrate the `namespace` field to a single-value `namespaces`? Can't the current 'single namespace' objects just be migrated to be multi-namespaced? Or are there other technical or functional implications between single and multi namespaced objects?

Reply:
> I guess this is all but easy, but having these two fields sounds and the distinction between single and multi namespaces SOs sounds wrong to me. How hard and breaking would be to migrate the `namespace` field to a single-value `namespaces`? Can't the current 'single namespace' objects just be migrated to be multi-namespaced? Or are there other technical or functional implications between single and multi namespaced objects?

I haven't fully thought through this yet, but I don't think we have a technical reason for keeping these two distinct fields (@jportner keep me honest here). That being said, the migration of `namespace` to `namespaces` would be a large undertaking, in part because the current migration system does not allow for such transformations. We have rather ambitious migration goals for subsequent phases of the Sharing Saved Objects initiative, some of which are being discussed in https://github.com/elastic/kibana/issues/59598.
We're considering all migration work out of scope for this initial phase.

Reply:
Yes, this PR is already pretty large and migration is out of scope right now. There are three more phases before we could migrate all of the saved object types, this is described in #27004.